src/irc/IdentGenerator.ts

```diff
@@ -43,6 +44,13 @@ export class IdentGenerator {
         });
     }
 
+    static switchAroundMxid(user: MatrixUser) {
+        return user.host.split('.')
+                .reverse()
+                .join('.')
+                .substr(0, 30) + (user.host.length > 30 ? ">:" : ":") + user.localpart;
```

Comment:
`.substr` is not an ECMAScript standard.
```suggestion
                .substring(0, 30) + (user.host.length > 30 ? ">:" : ":") + user.localpart;
```

Reply:
I wish the linter told me not to do this, thanks!